repository.go

```diff
@@ -546,49 +548,25 @@ func cleanUpDir(path string, all bool) error {
 
 // Config return the repository config. In a filesystem backed repository this
 // means read the `.git/config`.
-func (r *Repository) Config() (*config.Config, error) {
-	return r.Storer.Config()
+func (r *Repository) Config(o ...config.WithOptions) (*config.Config, error) {
+	return config.LocalScope.LoadFromConfigStorer(r.Storer, o...)
 }
 
-// SetConfig marshall and writes the repository config. In a filesystem backed
-// repository this means write the `.git/config`. This function should be called
-// with the result of `Repository.Config` and never with the output of
-// `Repository.ConfigScoped`.
+// SetConfig marshall and overwrites the local repository config. In a filesystem
+// backed repository this means write/overwrite the `.git/config` file. This function
+// should only be called with the result of `Repository.Config` or
+// `Repository.ConfigScoped(config.LocalScopeV6)` and never with the output of
+// any other `Repository.ConfigScoped`.
 func (r *Repository) SetConfig(cfg *config.Config) error {
 	return r.Storer.SetConfig(cfg)
 }
 
 // ConfigScoped returns the repository config, merged with requested scope and
 // lower. For example if, config.GlobalScope is given the local and global config
 // are returned merged in one config value.
-func (r *Repository) ConfigScoped(scope config.Scope) (*config.Config, error) {
+func (r *Repository) ConfigScoped(scope config.Scope, o ...config.WithOptions) (*config.Config, error) {
```

Comment:
Same here on the backwards compatibility. It is always hard to know how folks are using our public API, and changing it on a minor release would be problematic.